core-bundle/src/EventListener/LegacyLoginConstantsListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+/**
+ * @internal
+ *
+ * @deprecated Deprecated since Contao 4.9, to be removed in Contao 5.0
+ */
+class LegacyLoginConstantsListener
+{
+    /**
+     * @var TokenChecker
+     */
+    private $tokenChecker;
+
+    /**
+     * @var ScopeMatcher
+     */
+    private $scopeMatcher;
+
+    public function __construct(TokenChecker $tokenChecker, ScopeMatcher $scopeMatcher)
+    {
+        $this->tokenChecker = $tokenChecker;
+        $this->scopeMatcher = $scopeMatcher;
+    }
+
+    public function __invoke(RequestEvent $event): void
+    {
+        if ($this->scopeMatcher->isFrontendRequest($event->getRequest())) {
+            \define('BE_USER_LOGGED_IN', $this->tokenChecker->hasBackendUser() && $this->tokenChecker->isPreviewMode());
```

Comment:
be aware that in upstream-merge we must remove the `hasBackendUser` because in 4.13 a user can be in preview mode without being backend-authenticated (through preview links)